seq2science/rules/configuration_generic.smk

```diff
@@ -332,259 +330,54 @@ else:
-                             f"and since the sample did not start with either GSM, SRR, ERR, and DRR we couldn't find it online..\n")
-
-    # now parse the output and store the cache, the local files' layout, and the ones that were fetched online
-    config['layout'] = {**layout_cache,
-                        **{k: v for k, v in config['layout'].items()},
-                        **{k: v.get() for k, v in eutils_layout.items() if v.get() is not None},
-                        **{k: v.get()[0][0] for k, v in trace_layout1.items() if v.get() is not None},
-                        **{k: v.get()[0][0] for k, v in trace_layout2.items() if v.get() is not None}}
-
-    assert all(layout in ['SINGLE', 'PAIRED'] for sample, layout in config['layout'].items())
-
-    # if new samples were added, update the cache
-    if len([sample for sample in samples.index if sample not in layout_cache]) != 0:
-        pickle.dump({**config['layout']}, open(layout_cachefile, "wb"))
-
-
-# now only keep the layout of samples that are in samples.tsv
-config['layout'] = {**{key: value for key, value in config['layout'].items() if key in samples.index},
-                    **{key: value for key, value in config['layout'].items() if "control" in samples and key in samples["control"].values}}
-
-bad_samples = [sample for sample in samples.index if sample not in config["layout"]]
-if len(bad_samples) > 0:
-    logger.error(f"\nThe instructions to lookup sample(s) {' '.join(bad_samples)} online failed!\n"
-                 f"Are you sure these sample(s) exists..? Downloading samples with restricted "
-                 f"access is currently not supported. We advise you to download the sample "
-                 f"manually, and continue the pipeline from there on.\n")
-    raise TerminatedException
+        sampledict = {}
 
+    missing_samples = [sample for sample in all_samples if sample not in sampledict.keys()]
+    if len(missing_samples) > 0:
+        sampledict.update(samples2metadata(missing_samples, config))
 
-sample_to_srr = {**{k: [(config["layout"][k], k)] for k in trace_layout1.keys() if k.startswith("SRR")},
-                 **{k: v.get() for k, v in trace_layout1.items() if v.get() is not None},
-                 **{k: v.get() for k, v in trace_layout2.items() if v.get() is not None}}
+    pickle.dump(sampledict, open(eutils_cache, "wb"))
 
-trace_tp.close()
-eutils_tp.close()
+    # only keep samples for this run
+    sampledict = {sample: values for sample, values in sampledict.items() if sample in all_samples}
 
 logger.info("Done!\n\n")
-logger.info("Now checking if the samples are on the ENA database..")
-logger.info("This can also take some time!")
-
-ena_single_end_urls = dict()
-ena_paired_end_urls = dict()
 
-# trace_tp = ThreadPool(40)
+# now check where to download which sample
+ena_protocol = "ena_fastq_ftp" if config.get("ascp_path") and config.get("ascp_key") else "ena_fastq_http"
```

Comment:
wget should work on both ftp and http, and speed is similar. This means we can set ftp as default, and fall back to http if it doesnt exist/work. This is not an immediate requirement tho.